docs/docs/queries/structure.md

```diff
@@ -1,44 +1,207 @@
+- zero or one **FROM** data command with one to many [sources](../reference/sources.md)
+- zero to many other **data commands** with one to many [expressions](../reference/expressions.md) and/or other infos depending on the data command 
 
-The general format for queries is:
+Abstractly speaking, a query conforms the following pattern:
 
 ~~~
 ```dataview
-TABLE|LIST|TASK <field> [AS "Column Name"], <field>, ..., <field> 
+<QUERY-TYPE> <fields>
 FROM <source>
-WHERE <expression>
-SORT <expression> [ASC/DESC]
-... other data commands
+<DATA COMMAND> <expression>
 ```
 ~~~
 
-Only the [**Query Type**](./query-types.md) (table/list/task/calendar) statement is required - if the "from" statement is omitted, the query runs for all files in
-your vault. You can specify [data commands](./data-commands.md) like `WHERE` multiple times; they will run in the order they are written. 
+!!! hint "Only the Query Type is mandatory."
+
+The following sections will walk through the theory in more detail.
+
+## Choose a Output Format
+
+The final output appareance of a query is determined by its **Query Type**. There are four available:
+
+1. **TABLE**: The traditional view type; one row per result, with several columns of field data.
+2. **LIST**: A list of pages which match the query. You can output one field for each page alongside their file links.
+3. **TASK**: A list of tasks that match the given query.
+4. **CALENDAR**: A calendar view displaying each hit via a dot on its referred date.
 
-The most basic example of a dataview query is:
+The Query Type is the **only mandatory command in a query**. Everything else is optional.
 
 ~~~
+Lists all pages in your vault as a table with one column containing the page link
+```dataview
+TABLE
+```
+
+Lists all pages in your vault as a bullet point list
 ```dataview
 LIST
 ```
+
+Lists all tasks (completed or not) in your vault
+```dataview
+TASK
+```
+
+Renders a Calendar view where each page is represented as a dot on its creation date.
+```dataview
+CALENDAR file.cday
+```
 ~~~
 
-which will list **all files in your vault**. Find more examples [here](../resources/examples.md).
\ No newline at end of file
+!!! info "Read more about the available types and how to use them [here](./query-types.md)."
+
+## Choose your source
+
+Additionally to the Query Types, you have several **Data Commands** available that help you restrict, refine, sort or group your query. One of these query commands is the **FROM** statement, which is a bit special.
+
+!!! info "`FROM` determines which set of notes should be included in your query depending on given [sources](../../reference/sources)."
+
+You can add **zero or one** `FROM` data command to your query, right after your Query Type. You cannot add multiple FROM statements and you cannot add it after other Data Commands.
+
+`FROM` takes [sources](../../reference/sources) as argument and **restricts the pages your query will collect to the given sources**.
```

Comment:
FROM takes a source or a combination of sources as an argument. It restricts the query to a set of pages that match your source. This is important to keep in mind for query optimization.

Something along these lines would be useful IMO but it's not really necessary I think.